packages/web/pages/dashboard/profile/[id]/index.tsx

```diff
@@ -1,41 +1,58 @@
 import React from 'react'
-import { NextPage } from 'next'
 import { useRouter } from 'next/router'
 
 import { withApollo } from '@/lib/apollo'
-import LoadingWrapper from '@/components/LoadingWrapper'
-import DashboardLayout from '@/components/Layouts/DashboardLayout'
-import { useProfilePageQuery } from '@/generated/graphql'
-import useUILanguage from '@/hooks/useUILanguage'
-import Profile from '@/components/Dashboard/Profile'
-
-interface InitialProps {
-  namespacesRequired: string[]
-}
+import { useUserByIdentifierQuery } from '@/generated/graphql'
+import theme from '@/theme'
 
-const ProfilePage: NextPage<InitialProps> = () => {
-  const idStr = useRouter().query.id as string
+const ProfilePage = () => {
+  const router = useRouter()
+  const idStr = router.query.id as string
   const userId = parseInt(idStr, 10)
 
-  const uiLanguage = useUILanguage()
-  const { data, loading, error } = useProfilePageQuery({ variables: { uiLanguage, userId } })
-
-  const { userById, currentUser } = data || {}
-  const posts = data?.posts?.posts
-
-  return (
-    <LoadingWrapper loading={loading} error={error}>
-      <DashboardLayout pad="never">
-        {userById && posts && (
-          <Profile isLoggedInUser={currentUser?.id === userId} user={userById} posts={posts} />
-        )}
-      </DashboardLayout>
-    </LoadingWrapper>
-  )
+  const { data, loading, error } = useUserByIdentifierQuery({ variables: { id: userId } })
+
+  if (data?.userByIdentifier && typeof window !== 'undefined') {
```

Comment:
This approach requires the client load the full page before doing a redirect. I think we can do some magic in getInitialProps to redirect via status code which would make this a bit clearer IMO. We can merge like this, but that is a follow up I’d like to do